### ISSUE TITLE
Always update projects if they are out of date

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#1014]
Currently the default is to **not** update projects if they are up to date. Actually I don't see a compelling reason to not make it defaulting to always update the project if it is out-of-date as there is not much a user can/want do anyways.

**Comment by HannesWell** [comment]
You are referring to the option marked in red, aren't you? IIRC there was some discussion in the past to enable it by default (and remove the experimental label). And too me that seem reasonable.

![grafik](https://user-images.githubusercontent.com/44067969/199089179-5e1d0737-2dff-4f0f-ad69-96ad37fa9f69.png)

**Comment by laeubi** [comment]
Yes, this is off by default so I would like to enable it by default and remove the experimental. If there are any issues we should fix them. Its really useless to have this disabled but show an error marker that tells you you should update the project...

**Comment by HannesWell** [comment]
> Yes, this is off by default so I would like to enable it by default and remove the experimental. If there are any issues we should fix them. Its really useless to have this disabled but show an error marker that tells you you should update the project...

Agree. Actually I always set that to enabled. But nevertheless I saw that warning/error in a workspace. But it could be that I forgot to set that option in the workspace. I have to check that when I'm in the office next time this week.

**Comment by mickaelistria** [comment]
I think it was marked experimental a long time ago; but many bugfixes and perf improvements have been implemented since then. So this feature is mature and -as far as I know- working well enough to be turned on by default
So overall, I fully support the proposal.

**Comment by mickaelistria** [comment]
Submitted https://github.com/eclipse-m2e/m2e-core/pull/1029
